### ISSUE TITLE
Create Method Quick fix missing

### ISSUE THREAD

**Issue by Crucialjun** [Dart-Code/Dart-Code#5370]
**Describe the bug**

 The create method quick fix is missing in the current update VS code extension

**To Reproduce**
Update to the latest Dart extension, create a method name for a method that has not been created 

**Expected behavior**
Show a create method quick fix if the method does not exist 

**Screenshots**
![image](https://github.com/user-attachments/assets/a9d0df51-bf57-4cc8-82fc-cb6d93887a7f)

**Please complete the following information:**

You can run the **Dart: Collect Diagnostic Information** command from the VS Code command palette (`F1`) to easily capture this information or provide it manually.

 - Operating System and version:
 - VS Code version: win 1.96.0
 - Dart extension version: 3.102.0
 - Dart/Flutter SDK version: Dart 3.6.0  / Flutter 3.27.0
 - Target device (if the issue relates to Flutter debugging):

**Comment by FMorschel** [comment]
Hi! @Crucialjun, can you reproduce this on an empty project and place the code here? Trying to repro from your screenshot is difficult since the pop-up covered some of the code (this is actually something that comes from Dart-SDK but we can look at it here, I think).

You can post code here by adding 3 backsticks ` on a line before with _dart_ and ending on a new line with more 3 backsticks.

**Comment by DanTup** [comment]
Yeah, it's not clear how to reproduce this - the name shown in the fix (`_changeSelectedIndex`) is not visible in the code. If you can provide code we can copy/paste that triggers the issue, it will be easier to debug. Thanks!

**Comment by FMorschel** [comment]
@Nana-Kwame-bot can you post a reproduction please?

It was not removed on purpose, it is probably a bug. You can help us fix it by giving a small repro on an empty project so we can see the problem.

**Comment by FMorschel** [comment]
I don't think it was showing up before without the `()` (unless when given to a function/variable where a function was expected). But I'll be sure to test it all (in older versions too) and will come back here with more info when I do. Thanks!

**Comment by FMorschel** [comment]
@Nana-Kwame-bot I've tested as far as 3.22.3 and no version shows the fix for creating a method unless you have written `_showDialog();` (notice the `()` in the end tell the analyzer that this is a function call).

I've also taken the liberty to suppose the original request was from some code similar to:

```dart
import 'package:bloc/bloc.dart';

class Class extends Bloc<int, int> {
  Class() : super(0) {
    on<int>(_changeSelectedIndex);
  }
}
```

And that does show the fix for me (since inside `on` it is expecting `FutureOr<void> Function(E, Emitter<int>) handler` which is a `Function`). I think we still need some info to track down what you are struggling with.

**Comment by Crucialjun** [comment]
I found a device i have not updated for sometime with the option still showing

![Screenshot 2024-12-20 at 22 20 32](https://github.com/user-attachments/assets/17d90bcb-c391-4c18-81bd-9558a6c7b615)

┌─────────────────────────────────────────────────────────┐
│ A new version of Flutter is available!                  │
│                                                         │
│ To update to the latest version, run "flutter upgrade". │
└─────────────────────────────────────────────────────────┘
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.0 23A5257q darwin-x64, locale en-US)
    • Flutter version 3.24.3 on channel stable at /Users/nich_otieno/Downloads/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (3 months ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).

[✓] VS Code (version 1.95.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.101.20241031

[✓] Connected device (2 available)            
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.0 23A5257q darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 131.0.6778.205

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

**Comment by tranvu1805** [comment]
it was occurred to me, i cannot create method in bloc.
![Uploading Screenshot 2024-12-21 111256.png…]()

**Comment by FMorschel** [comment]
@Crucialjun and @tranvu1805 can any of you please inform me what is the version of Bloc on `pubspec.lock` for the machines that don't work?

**Comment by Crucialjun** [comment]
On the machine that doesnt have the option , the flutter_bloc version is 8.1.6

**Comment by FMorschel** [comment]
@Crucialjun @Nana-Kwame-bot @tranvu1805

I could repro the problem on the latest stable version but not on Dart master. I think it has already been solved and will ship on a future Dart SDK release. I'll see if I can find what commit fixed it and post it here.

**Comment by FMorschel** [comment]
The only commits I could find around the time of the release that should change anything for this fix are mine but when I worked on them the fix was working for this case you showed.

- https://github.com/dart-lang/sdk/commit/a0374dd70cd460aca08b0a1f91637bf3fe7c48d5 (https://github.com/dart-lang/sdk/issues/56719)
- https://github.com/dart-lang/sdk/commit/06111a51c55248f0111118a0ab37f92ae7188406 (https://github.com/dart-lang/sdk/issues/58266)

Not still sure what broke this, will keep looking when I have some time and I'll take a look if we have a test for this case to make sure it won't break again on the future.

Things are a bit slow now at the end of the year since the Dart team is taking a rest (well deserved vacation). Once the new year starts things will come back to the normal pace. In the meantime I suggest you all to wait for a new version of the SDK. I'm not sure if we ever make cherry-picks for non-breaking things like this, but I'll ask them about this.

CC @DanTup read this comment and the above, please. Thanks!

**Comment by DanTup** [comment]
I can reproduce this on stable. `git bisect` led me to c38244e7b62c5cf5c7a27d5d11651e3b8558cd3e, and indeed if I revert the change to `correspondingParameter` from that change (so it doesn't handle `ParameterMember`), then it's broken on main too.

So, I think we can consider this fixed (by c38244e7b62c5cf5c7a27d5d11651e3b8558cd3e). Thanks!

**Comment by TheManuz** [comment]
I tried version `3.105.20250207` (pre-release) but the behavior is still not fixed.

I'm trying to find the last working version with a working "Create method" quickfix, but I cannot find it.

When can we expect to see this fix released?

**Comment by FMorschel** [comment]
@TheManuz and @Crucialjun the fix for this is not in Dart-Code. This is on the bundled Dart SDK inside your Flutter SDK (or your standalone Dart SDK if you're not using Flutter). The latest master definitely has this covered but the Beta for 3.7.0+ also has this fix. Please try that and let me know if this is fixed. You can download it here: https://dart.dev/get-dart/archive

**Comment by FMorschel** [comment]
To be very clear:

If you're using Flutter, moving between the Beta channel and Stable channel should be easy enough and it would have the fix too. 

Downloading the standalone Dart SDK is not recommended in this case since you'd need to replace the built-in Dart in Flutter SDK (not even sure it won't break anything if you don't know what you're doing). Downloading the Dart SDK on the link I suggested above is the path for Dart only users.

**Comment by Crucialjun** [comment]
Just tested on 3.29.0-0.3.pre and confirmed to be working

**Comment by DanTup** [comment]
[Flutter 3.29](https://medium.com/flutter/whats-new-in-flutter-3-29-f90c380c2317) and [Dart 3.7](https://medium.com/dartlang/announcing-dart-3-7-bf864a1b195c) have been released to the stable channels, so if you upgrade to latest you should have this fix on stable now :-)

**Comment by andynvt** [comment]
> [Flutter 3.29](https://medium.com/flutter/whats-new-in-flutter-3-29-f90c380c2317) and [Dart 3.7](https://medium.com/dartlang/announcing-dart-3-7-bf864a1b195c) have been released to the stable channels, so if you upgrade to latest you should have this fix on stable now :-)

I think it still occurs on 3.29. Previously, it has: "create function X" to quickly create a void/Future function from that point

![Image](https://github.com/user-attachments/assets/63d9f229-7118-4fb4-ab78-55900bcd5f0c)

**Comment by FMorschel** [comment]
Hey @andynvt, please open another issue at https://github.com/dart-lang/sdk/issues. If you can please create a small example of this behavior containing the least ammout of code, we'd appreciate it.

This is not exactly the same as the above (at least on how it is implemented internally). Please also tag me so I can follow up on your case there too. 

To help you with the reproduction: this exact case `Create extension method`, if I remember correctly is only triggered on classes that are not from the current package, but I can be wrong. If `model` is from a type outside your project, this is probably why. You can't `Create method` into types/classes you don't own.
